model/buildlogger.go

```diff
@@ -217,9 +218,49 @@ func (l *Log) appendLogChunkInfo(logChunk LogChunkInfo) error {
 	if err == nil && updateResult.MatchedCount == 0 {
 		err = errors.Errorf("could not find log record with id %s in the database", l.ID)
 	}
+
 	return errors.Wrapf(err, "problem appending log chunk info to %s", l.ID)
 }
 
+// CloseLog "closes out" the log by populating the completed_at and
+// info.exit_code fields. It should be the last call made on a buildlogger log.
+func (l *Log) CloseLog(completedAt time.Time, exitCode int) error {
+	if !l.populated {
+		return errors.New("cannot close log when log unpopulated")
+	}
+	if l.env == nil {
+		return errors.New("cannot close log with a nil environment")
+	}
+	ctx, cancel := l.env.Context()
+	defer cancel()
+
+	if l.ID == "" {
+		l.ID = l.Info.ID()
+	}
+
+	update := bson.M{
+		"$set": bson.M{
+			logCompletedAtKey: completedAt,
+			bsonutil.GetDottedKeyName(logInfoKey, logInfoExitCodeKey): exitCode,
+		},
+	}
+	updateResult, err := l.env.GetDB().Collection(buildloggerCollection).UpdateOne(ctx, bson.M{"_id": l.ID}, update)
+	grip.DebugWhen(err == nil, message.Fields{
+		"ns":           model.Namespace{DB: l.env.GetConf().DatabaseName, Collection: buildloggerCollection},
```

Comment:
I'd just put the collection. I think GetConf is (or could be in the future) more expensive than we'd like, and the db name isn't as confusing.

Reply:
Set it to just the collection for all of these debug statements. Should I do the same in perf?

Reply:
yeah